### PR TITLE
App dir bug #201

### LINKED ISSUE #201: Default app directory path is null (CF CLI can't find app to push unless path explicitly specified)
In the absence of a manifest, the deployment dialog uses `null` for the value of the app path. The CloudFoundryService is currently looking up the `path` value in the provided manifest & using that as the "working directory" when it executes the `cf push` command; when that value is null, the CF CLI doesn't know where the app bits live that it's supposed to push.

### PR DESCRIPTION
**Pull request by atwoosnam**
fixes #201

### CODE REVIEW

File: src/Services/src/CloudFoundry/CloudFoundryService.cs [hunk 1]
```diff
@@ -774,11 +774,11 @@ public async Task<DetailedResult> DeleteAppAsync(CloudFoundryApp app, bool skipS
             };
         }
 
-        public async Task<DetailedResult> DeployAppAsync(AppManifest appManifest, CloudFoundryInstance targetCf, CloudFoundryOrganization targetOrg, CloudFoundrySpace targetSpace, StdOutDelegate stdOutCallback, StdErrDelegate stdErrCallback)
+        public async Task<DetailedResult> DeployAppAsync(AppManifest appManifest, string defaultAppPath, CloudFoundryInstance targetCf, CloudFoundryOrganization targetOrg, CloudFoundrySpace targetSpace, StdOutDelegate stdOutCallback, StdErrDelegate stdErrCallback)
         {
             AppConfig app = appManifest.Applications[0];
 
-            string pathToDeploymentDirectory = app.Path;
+            string pathToDeploymentDirectory = app.Path == null ? defaultAppPath : app.Path;
```
Comment (ccheetham) [inline src/Services/src/CloudFoundry/CloudFoundryService.cs:781]:
```suggestion
            string pathToDeploymentDirectory = app.Path ?? defaultAppPath;
```

File: src/Services/test/CloudFoundry/CloudFoundryServiceTests.cs [hunk 2]
```diff
@@ -1708,6 +1708,70 @@ public async Task DeployAppAsync_ReturnsFailedResult_WhenCfCliDeploymentThrowsIn
...
+        {
+            var appManifest = new AppManifest()
+            {
+                Applications = new List<AppConfig>
+                {
+                    new AppConfig(),
+                }
+            };
+
+            var defaultAppPath = "fake\\app\\path";
```
Comment (ccheetham) [inline src/Services/test/CloudFoundry/CloudFoundryServiceTests.cs:1723]:
```suggestion
            var defaultAppPath = @"fake\app\path";
```

File: src/Services/test/CloudFoundry/CloudFoundryServiceTests.cs [hunk 3]
```diff
@@ -1708,6 +1708,70 @@ public async Task DeployAppAsync_ReturnsFailedResult_WhenCfCliDeploymentThrowsIn
...
+        [TestCategory("DeployApp")]
+        public async Task DeployAppAsync_UsesManifestAppPath_WhenManifestAppPathIsNotNull()
+        {
+            var appManifest = new AppManifest()
+            {
+                Applications = new List<AppConfig>
+                {
+                    new AppConfig()
+                    {
+                        Path = "manifest\\app\\path",
```
Comment (ccheetham) [inline src/Services/test/CloudFoundry/CloudFoundryServiceTests.cs:1751]:
```suggestion
                        Path = @"manifest\app\path",
```

**Review by ccheetham — CHANGES_REQUESTED** [review]
syntactical changes

File: src/Services/test/CloudFoundry/CloudFoundryServiceTests.cs (hunk 3 above)
Comment (atwoosnam) [reply src/Services/test/CloudFoundry/CloudFoundryServiceTests.cs]: Nice! Done in ba32832

**Pull request merged**